### ISSUE TITLE
Neighbor Advertisement Not Returned for duplicate address detect (DAD)

### ISSUE THREAD

**Issue by cynkiller** [cilium/cilium#18928]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

On Suse environment with k8s 1.21 and cilium v1.9.8 installed, if I set the same ipv6 address on other node in the same layer-2 broadcast zone, the node can receive neighbor solicitation packet, but the neighbor advertisement packet is dropped by cilium and so it doesn't return NA packet to :: , so that the address duplication cannot be detected.
When I directly send neighbor solicitation packet to the target node, the neighbor advertisement result can be properly return. So that this issue only happens for default routing address :: 

The ipv6 address on this node is fc00::20:16:10:200
```
# ifconfig eth0
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.16.10.200  netmask 255.255.0.0  broadcast 172.16.255.255
        inet6 fc00::20:16:10:200  prefixlen 96  scopeid 0x0<global>
        inet6 fe80::20c:29ff:fe17:7f02  prefixlen 64  scopeid 0x20<link>
        ether 00:0c:29:17:7f:02  txqueuelen 1000  (Ethernet)
        RX packets 6022  bytes 665241 (649.6 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 1987  bytes 379113 (370.2 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
# ifconfig cilium_host
cilium_host: flags=4291<UP,BROADCAST,RUNNING,NOARP,MULTICAST>  mtu 1500
        inet 172.28.175.106  netmask 255.255.255.255  broadcast 0.0.0.0
        inet6 fe80::50aa:92ff:fe41:5816  prefixlen 64  scopeid 0x20<link>
        inet6 fc00::20:16:10:200  prefixlen 128  scopeid 0x0<global>
        ether 52:aa:92:41:58:16  txqueuelen 1000  (Ethernet)
        RX packets 5  bytes 550 (550.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 11  bytes 1062 (1.0 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

tcpdump result:
```
<<< unicast NS, NA returned
03:08:35.021668 IP6 fe80::250:56ff:fe98:8308 > ff02::1:ff10:200: ICMP6, neighbor solicitation, who has fc00::20:16:10:200, length 32
03:08:35.021685 IP6 fd83:8383:8383::6397 > fe80::250:56ff:fe98:8308: ICMP6, neighbor advertisement, tgt is fc00::20:16:10:200, length 32
<<< multicast NS, NA not returned
03:09:00.617580 IP6 :: > ff02::1:ff10:200: ICMP6, neighbor solicitation, who has fc00::20:16:10:200, length 24
```

cilium monitor result:
```
<<< multicast NA dropped for invalid packet
xx drop (Invalid packet) flow 0xc007e92c to endpoint 0, identity 0->0: :: -> ff02::1:ff10:200 NeighborAdvertisement
```

### Cilium Version

v1.9.8

### Kernel Version

```
# uname -a
Linux localhost 5.3.18-24.96-default #1 SMP Mon Nov 22 08:33:36 UTC 2021 (9a413cc) x86_64 x86_64 x86_64 GNU/Linux
```

### Kubernetes Version

v1.21.4+rke2r2
```
# kubectl version
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.4+rke2r2", GitCommit:"3cce4a82b44f032d0cd1a1790e6d2f5a55d20aae", GitTreeState:"clean", BuildDate:"2021-08-29T22:29:54Z", GoVersion:"go1.16.6b7", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.4+rke2r2", GitCommit:"3cce4a82b44f032d0cd1a1790e6d2f5a55d20aae", GitTreeState:"clean", BuildDate:"2021-08-29T22:29:54Z", GoVersion:"go1.16.6b7", Compiler:"gc", Platform:"linux/amd64"}
```

### Relevant log output

```shell
xx drop (Invalid packet) flow 0xc007e92c to endpoint 0, identity 0->0: :: -> ff02::1:ff10:200 NeighborAdvertisement
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by cynkiller** [comment]
Is there any update on this issue? Thanks.

**Comment by borkmann** [comment]
> Is there any update on this issue? Thanks.

Hello @cynkiller, do you experience the issue also with latest 1.11 stable?

**Comment by andy-v-h** [comment]
> > Is there any update on this issue? Thanks.
> 
> Hello @cynkiller, do you experience the issue also with latest 1.11 stable?
Not sure if i'm seeing a related issue or the same one but, I'm running latest stable 1.11 and seeing these kinds of drops consisently

```
xx drop (Unsupported L3 protocol) flow 0xf0475ef6 to endpoint 0, , identity unknown->unknown: 2604::6 -> ff02::1 NeighborAdvertisement
xx drop (Unsupported L3 protocol) flow 0x5b06d9ec to endpoint 0, , identity unknown->unknown: fe80::deff:fead:beef -> ff02::1 NeighborAdvertisement
xx drop (Unsupported L3 protocol) flow 0xf0475ef6 to endpoint 0, , identity unknown->unknown: 2604::6 -> ff02::1 NeighborAdvertisement
xx drop (Unsupported L3 protocol) flow 0x5b06d9ec to endpoint 0, , identity unknown->unknown: fe80::deff:fead:beef -> ff02::1 NeighborAdvertisement
xx drop (Unsupported L3 protocol) flow 0x0 to endpoint 0, , identity 24627->unknown: fe80::55ff:feca:9b1d -> ff02::2 RouterSolicitation
xx drop (Unsupported L3 protocol) flow 0x0 to endpoint 0, , identity 7065->unknown: fe80::3eff:fe79:1a94 -> ff02::2 RouterSolicitation
```

**Comment by frankui** [comment]
@borkmann We also observed this issue with cilium 1.11.2.
DAD multicaset NA was dropped by cilium which blocked the DAD of other IPv6 devices.

**Comment by msune** [comment]
This _might_ get fixed with #36024.

TL;DR, NS for DAD don't have LLSRC option, but before #36024 the BPF DP code was expecting it. See: https://github.com/cilium/cilium/pull/36024/commits/05fb7be77f3d86a4a63bb8986bf21143598e3f12

**Comment by msune** [comment]
I believe this issue should be fixed by now with:

* https://github.com/cilium/cilium/pull/36024
 
so I would close it.
